include/cantera/clib/ct.h

```diff
@@ -65,8 +65,8 @@ extern "C" {
     CANTERA_CAPI int thermo_setName(int n, const char* nm);
     CANTERA_CAPI size_t thermo_elementIndex(int n, const char* nm);
     CANTERA_CAPI size_t thermo_speciesIndex(int n, const char* nm);
-    CANTERA_CAPI int thermo_report(int nth,
-                                  int ibuf, char* buf, int show_thermo);
+    //! @since Changed signature in %Cantera 3.1
+    CANTERA_CAPI int thermo_report3(int nth, int show_thermo, double threshold, int ibuf, char* buf);
```

Comment:
So, taking the approach of not requiring `clib` to have a stable API, we should be able to change the argument order without having to rename the function. Anyone using this function will get a compiler error if they try to call the function with the old argument order.

Likewise for `ct_getDataDirectories3`.